docs/usage/administration/admin_api/federation.md

```diff
@@ -0,0 +1,81 @@
+# Federation API
+
+This API allows a server administrator to manage Synapse's federation with other homeservers.
+
+Note: This API is new, experimental and "subject to change".
+
+## List of destinations
+
+This API gets the current destination retry timing info for a remote server.
+
+The API is:
+
+A standard request with no filtering:
+
+```
+GET /_synapse/admin/v1/federation/destinations
+```
+
+A response body like the following is returned:
+
+```json
+{
+   "destinations":[
+      {
+         "destination": "matrix.org",
+         "retry_last_ts": 1557332397936,
+         "retry_interval": 3000000,
+         "failure_ts": 1557329397936,
+         "last_successful_stream_ordering": null
+      }
+   ],
+   "total": 1
+}
+```
+
+To paginate, check for `next_token` and if present, call the endpoint again
+with `from` set to the value of `next_token`. This will return a new page.
+
+If the endpoint does not return a `next_token` then there are no more destinations
+to paginate through.
+
+**Parameters**
+
+The following query parameters are available:
+
+- `from` - Offset in the returned list. Defaults to `0`.
+- `limit` - Maximum amount of destinations to return. Defaults to `100`.
+- `order_by` - The method in which to sort the returned list of destinations.
+  Valid values are:
+  - `destination` - Destinations are ordered alphabetically by remote server name.
+    This is the default.
+  - `retry_last_ts` - Destinations are ordered by time of last retry attempt in ms.
+  - `retry_interval` - Destinations are ordered by how long until next retry in ms.
+  - `failure_ts` - Destinations are ordered by when the server started failing in ms.
+  - `last_successful_stream_ordering` - Destinations are ordered by the stream ordering
+    of the most recent successfully-sent PDU.
+- `dir` - Direction of room order. Either `f` for forwards or `b` for backwards. Setting
+  this value to `b` will reverse the above sort order. Defaults to `f`.
+
+*Caution:* The database only has an index on the column `destination`.
+This means that if a different sort order is used,
+this can cause a large load on the database, especially for large environments.
+
+**Response**
+
+The following fields are returned in the JSON response body:
```

Comment:
A couple of questions:

- What happens if I'm federating with a server and federation has never failed? Does it appear in this list?
- What happens if federation with a problematic server recovers---does it appear in this list?
- Does the data returned persist across Synapse restarts?
- I think someone told me that Synapse has some kind of "blocking" mechanism, where if a remote homeserver fails too many federation requests, we block all federation requests to that server. Is that reflected in this output at all?

Sorry---I appreciate that some of these questions are about how Synapse itself works, rather than this API. But I think they're the kind of thing that'd be useful to consider in the documentation.

Reply:
A few answers:
> - What happens if I'm federating with a server and federation has never failed? Does it appear in this list?

Every federated server should be in the list.
https://github.com/matrix-org/synapse/blob/2b82ec425fccb0ef626242779f7ccd4d77a0685c/synapse/storage/databases/main/transactions.py#L368-L369

> What happens if federation with a problematic server recovers---does it appear in this list?

Yes.
https://github.com/matrix-org/synapse/blob/2b82ec425fccb0ef626242779f7ccd4d77a0685c/synapse/storage/databases/main/transactions.py#L203-L204

> Does the data returned persist across Synapse restarts?

Yes.

> I think someone told me that Synapse has some kind of "blocking" mechanism, where if a remote homeserver fails too many federation requests, we block all federation requests to that server. Is that reflected in this output at all?

I don't know.